### DISCUSSION TITLE
issue: "errors" object is empty after first trigger

### DISCUSSION THREAD

**Discussion by benjaminbenais** [react-hook-form/react-hook-form#8421] [Q&A]
### Version Number

7.31.3

### Codesandbox/Expo snack

https://codesandbox.io/s/react-hook-form-v6-triggervalidation-forked-xxeckn?file=/src/index.js

### Steps to reproduce

1. Go to 'index.js'
2. Click on 'Validate all'
3. Check the console
4. The error object is empty
5. Click on 'Validate all' again
6. The object error contains the validation error

### Expected behaviour

Hi, when I check the [official example](https://codesandbox.io/s/react-hook-form-v6-triggervalidation-forked-8w9tn) (using v7.0.0) related to the 'trigger' method, I can get the 'errors' object right away when the fileds are triggered.
However, when I update from 7.0.0 to 7.31.3, the errors object is empty the first time I click on the 'Validate all' button, but populated with the errors the second time I click the same button.

Is this a normal behavior ? Any solution ? I really need to get the errors the first time a trigger is performed.

Also, the sandobox says v6 although v7 is installed.

Thanks !

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
`errors` will gets updated in the next render, move your `console.log` outside of the render.

**Reply by benjaminbenais** [reply]
Thanks for your answer. Can you provide an example ? I don't see where else to put the console.log.

I have a multi step form, I would like to trigger the errors when a "Next step" button is clicked. How can I get the errors when clicking a non-submit button to prevent from accessing the next step of the form ?

**Reply by benjaminbenais** [reply]
I've found a workarround using an additional state value binded to a useEffect, which is not ideal. The official example on codesandbox is somehow missleading since it doesn't reflect the most up to date version.
